tests/TestCompiler.hs

```diff
@@ -227,7 +227,9 @@ checkShouldFailWith expected errs =
   let actual = map P.errorCode $ P.runMultipleErrors errs
   in if sort expected == sort (map T.unpack actual)
     then Nothing
-    else Just $ "Expected these errors: " ++ show expected ++ ", but got these: " ++ show actual
+    else Just $ "Expected these errors: " ++ show expected ++ ", but got these: " 
+      ++ show actual ++ ", full error messages: " 
+      ++ unlines (map (P.renderBox . P.prettyPrintSingleError P.defaultPPEOptions) (P.runMultipleErrors errs))
```

Comment:
This modification to the reporting made it much easier to debug exactly what was going on with a mistaken error code.

Reply:
I'd be in favour of merging this in a separate PR.

Reply:
Cool, I'll factor it out then.